Tests/MongoSwiftTests/ChangeStreamTests.swift

```diff
@@ -14,153 +104,177 @@ final class ChangeStreamTest: MongoSwiftTestCase {
-            var replyPtr = UnsafeMutablePointer<BSONPointer?>.allocate(capacity: 1)
-            defer {
-                replyPtr.deinitialize(count: 1)
-                replyPtr.deallocate()
-            }
-            expect(try ChangeStream<ChangeStreamDocument<Document>>(stealing: changeStreamPtr,
-                                                                    client: client,
-                                                                    connection: connection,
-                                                                    session: session,
-                                                                    decoder: decoder))
-                                                                    .toNot(throwError())
-        }
+        let changeStream = try coll.watch(options: options)
+
+        // expect the change stream to contain a change document for the `insert` operation
+        try coll.insertOne(["x": 1])
+        let res1 = changeStream.next()
+        expect(res1).toNot(beNil())
+        expect(res1?.operationType).to(equal(.insert))
+        expect(res1?.fullDocument?["x"]).to(bsonEqual(1))
+
+        // expect the resumeToken to be updated to the _id field of the most recently accessed document
+        expect(changeStream.resumeToken).to(equal(res1?._id))
+
+        try coll.updateOne(filter: ["x": 1], update: ["$set": ["x": 2] as Document])
+        let res2 = changeStream.next()
+
+        // expect the change stream to contain a change document for the `update` operation
+        expect(res2).toNot(beNil())
+        expect(res2?.operationType).to(equal(.update))
+        expect(res2?.fullDocument?["x"]).to(bsonEqual(2))
+
+        // expect the resumeToken to be updated to the _id field of the most recently accessed document
+        expect(changeStream.resumeToken).to(equal(res2?._id))
+
+        // expect the change stream contains a change document for the `find` operation
+        try coll.findOneAndDelete(["x": 2])
+        let res3 = changeStream.next()
+        expect(res3).toNot(beNil())
+        expect(res3?.operationType).to(equal(.delete))
+
+        // expect the resumeToken to be updated to the _id field of the most recently accessed document
+        expect(changeStream.resumeToken).to(equal(res3?._id))
     }
 
-    func testChangeStreamOnADatabase() throws {
+    func testChangeStreamOnACollectionWithCodableType() throws {
         guard MongoSwiftTestCase.topologyType != .single else {
             print("Skipping test case because of unsupported topology type \(MongoSwiftTestCase.topologyType)")
             return
         }
 
         let client = try MongoClient()
+        let db = client.db(type(of: self).testDatabase)
+        defer { try? db.drop() }
 
-        guard try client.serverVersion() >= ServerVersion(major: 4, minor: 0) else {
-            print("Skipping test case for server version \(try client.serverVersion())")
+        struct MyType: Codable {
```

Comment:
consider making this type `Equatable`, then below you can just check if `res1.fullDocument` equals the exact instance you insert